drivers/console/ipm_console_receiver.c

```diff
@@ -66,51 +69,29 @@ static void ipm_console_thread(void *arg1, void *arg2, void *arg3)
 			++pos;
 		}
 
-		/* ISR may have disabled the channel due to full buffer at
-		 * some point. If that happened and there is now room,
-		 * re-enable it.
-		 *
-		 * Lock interrupts to avoid pathological scenario where
-		 * the buffer fills up in between enabling the channel and
-		 * clearing the channel_disabled flag.
-		 */
-		if (driver_data->channel_disabled &&
-		    ring_buf_space_get(&driver_data->rb)) {
-			key = irq_lock();
-			ipm_set_enabled(driver_data->ipm_device, 1);
-			driver_data->channel_disabled = 0;
-			irq_unlock(key);
-		}
 	}
 }
 
 static void ipm_console_receive_callback(void *context, u32_t id,
 					 volatile void *data)
 {
 	struct device *d;
-	struct ipm_console_receiver_runtime_data *driver_data;
+	struct ipm_console_receiver_runtime_data *volatile driver_data;
 	int ret;
+	k_spinlock_key_t key;
 
 	ARG_UNUSED(data);
 	d = context;
 	driver_data = d->driver_data;
 
+	key = k_spin_lock(&driver_data->rb_spinlock);
 	/* Should always be at least one free buffer slot */
 	ret = ring_buf_item_put(&driver_data->rb, 0, id, NULL, 0);
+	k_spin_unlock(&driver_data->rb_spinlock, key);
 	__ASSERT(ret == 0, "Failed to insert data into ring buffer");
 	k_sem_give(&driver_data->sem);
 
-	/* If the buffer is now full, disable future interrupts for this channel
-	 * until the thread has a chance to consume characters.
-	 *
```

Comment:
Same question as before: why is it OK to remove this?  The condition and the comment explaining it (a RX interrupt is out of space in the buffer) makes sense, and I don't see anything in the patch that addresses it.

Reply:
sender thread will wait until there is space in ring buffer to do next send.
+       while (ring_buf_space_get(&driver_data->rb) == 0) {
+       }